code-challenges/Object.keys, Object.values, Object.entries/challenges-06.test.js

```diff
@@ -0,0 +1,233 @@
+ 
+All of these objects should be added to an array named "sizes". Return the "sizes" array from the function.
+ 
+For example: [{ house: 'Stark', members: 7 }, { house: 'Arryn', members: 3 }, ... ].
+------------------------------------------------------------------------------------------------ */
+
+const houseSize = (arr) => {
+  const sizes = [];
+  // Solution code here...
+  return sizes;
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 7 - Stretch Goal
+ 
+As fans are well aware, "When you play the game of thrones, you win or you die. There is no middle ground."
+ 
+We will assume that Alerie Tyrell is deceased. She missed her daughter's wedding. Twice.
+ 
+Write a function named houseSurvivors. You may modify your houseSize function from challenge 6 to use as the basis of this function.
+ 
+This function should create an object for each house containing the name of the house and the number of members. If the spouse is deceased, do not include him/her in the total number of family members.
+ 
+All of these objects should be added to an array named "survivors". Return the "survivors" array from the function.
+ 
+For example: [ { house: 'Stark', members: 6 }, { house: 'Arryn', members: 2 }, ... ].
+------------------------------------------------------------------------------------------------ */
+
+const deceasedSpouses = ['Catelyn', 'Lysa', 'Robert', 'Khal Drogo', 'Alerie'];
+
+const houseSurvivors = (arr) => {
+  const survivors = [];
+  // Solution code here...
+  return survivors;
+};
+
+/* ------------------------------------------------------------------------------------------------
+TESTS
+ 
+All the code below will verify that your functions are working to solve the challenges.
+ 
+DO NOT CHANGE any of the below code.
+ 
+Run your tests from the console: jest challenges-06.test.js
+ 
+------------------------------------------------------------------------------------------------ */
+
+describe('Testing challenge 1', () => {
+  test('It should return the keys from an object', () => {
+    expect(getCourseKeys(courseInfo)).toStrictEqual(['name', 'duration', 'topics', 'finalExam']);
+  });
+});
+
+describe('Testing challenge 2', () => {
+  test('It should return an array of the names of the houses', () => {
+    expect(getHouses(characters)).toStrictEqual(['Stark', 'Arryn', 'Lannister', 'Targaryen', 'Tyrell', 'Greyjoy', 'Snow']);
+    expect(getHouses(characters).length).toStrictEqual(7);
+  });
+});
+
+describe('Testing challenge 3', () => {
+  test('It should return true for characters that have children', () => {
+    expect(hasChildrenValues(characters, 'Daenarys')).toBeTruthy();
+  });
+
+  test('It should return false to characters who do not have children', () => {
+    expect(hasChildrenValues(characters, 'Sansa')).toBeFalsy();
+  });
+});
+
+describe.only('Testing challenge 4', () => {
```

Comment:
Make sure to change this back to describe before you submit next time. Otherwise it skips all the other tests and I can't tell what answers you got right.